daml-assistant/src/DAML/Assistant/Version.hs

```diff
@@ -5,34 +5,33 @@
 
 module DAML.Assistant.Version
     ( getInstalledSdkVersions
-    , getLatestSdkVersionCached
     , getSdkVersionFromSdkPath
     , getSdkVersionFromProjectPath
     , getAssistantSdkVersion
     , getDefaultSdkVersion
+    , getAvailableSdkVersions
+    , getAvailableSdkVersionsCached
+    , refreshAvailableSdkVersions
+    , getLatestSdkVersionCached
     ) where
 
 import DAML.Assistant.Types
 import DAML.Assistant.Util
 import DAML.Assistant.Cache
-import DAML.Assistant.Install
 import DAML.Project.Config
 import DAML.Project.Consts hiding (getDamlPath, getProjectPath)
 import System.Directory
 import System.FilePath
 import System.Environment.Blank
+import Control.Exception.Safe
 import Control.Monad.Extra
 import Data.Maybe
+import Data.List
 import Data.Either.Extra
+import Data.Aeson (eitherDecodeStrict')
 import Safe
-
--- | Get the latest released SDK version. Designed to return Nothing if
--- anything fails (e.g. machine is offline). The result is cached in
--- $DAML_HOME/cache/latest-sdk-version.txt and only polled once a day.
-getLatestSdkVersionCached :: DamlPath -> IO (Maybe SdkVersion)
-getLatestSdkVersionCached damlPath =
-    cacheLatestSdkVersion damlPath $ do
-        tryAssistantM getLatestVersion
+import Network.HTTP.Simple
+import qualified Data.Map as M
```

Comment:
I thought Aeson would prefer to decode it to a Data.HashMap.Strict? And thus go the unordered-containers route?

Reply:
Didn't even think about it -- I'll switch over.